util/net/http_transport_mac.mm

```diff
@@ -289,6 +289,11 @@ static bool ExecuteProxyRequest(NSMutableURLRequest* request,
       @"HTTPSEnable" : @YES,
       @"HTTPSPort" : proxy_port,
       @"HTTPSProxy" : hostNS,
+    #if !TARGET_OS_IOS
```

Comment:
Since the Socks options are not available on iOS, we add this check. This gets checked at compile time, but I'm not sure if this is what we actually want (or how we'd check this at runtime instead)

Reply:
+ these options are available from MacOS 10.6 onwards; is this something we should check for too?

Reply:
> these options are available from MacOS 10.6 onwards; is this something we should check for too?

Nope. I think the lowest-mentioned support version of the Native SDK for MacOS is 10.15 (and even for this, I would wait until we get user feedback).

> Since the Socks options are not available on iOS, we add this check. This gets checked at compile time, but I'm not sure if this is what we actually want (or how we'd check this at runtime instead)

Are the `SOCKS` options unavailable on `iOS`, or are the constants just not defined in the `iOS` SDK? My guess is the latter and that string keys should work for both (similar to `HTTPS[Enable|Port|Proxy]`).

Reply:
By the way, I would expect a (runtime) conditional assignment to the `proxyDict`, right? 

Setting `HTTP` and `HTTPS` to configure an `HTTP` proxy makes sense because they refer to the target `URL`, not the proxy's protocol scheme. 

But now, would we assign the same server for `HTTP` and `SOCKS`? That seems ambiguous even if it "works".

Reply:
No, we could use string keys for all `proxyDict` elements to make it more consistent. No special case for iOS is necessary; it's just a stupidity in the SDK.